via/static/scripts/video_player/hooks/use-overflow-anchor.ts

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useRef } from 'preact/hooks';
+import type { MutableRef } from 'preact/hooks';
+
+function rectIsEmpty(rect: DOMRect) {
+  return rect.width <= 0 || rect.height <= 0;
+}
+
+/**
+ * Return true if `rectB` is fully contained within `rectA`
+ */
+export function rectContains(rectA: DOMRect, rectB: DOMRect) {
+  if (rectIsEmpty(rectA) || rectIsEmpty(rectB)) {
+    return false;
+  }
+
+  return (
+    rectB.left >= rectA.left &&
+    rectB.right <= rectA.right &&
+    rectB.top >= rectA.top &&
+    rectB.bottom <= rectA.bottom
+  );
+}
+
+type ChildInViewport = {
+  element: HTMLLIElement | null;
```

Comment:
`getChildren` returns a list of `HTMLElement`, which may not be an `li`, but the element is recorded as an `HTMLLIElement` here.

Reply:
Oh, this is an oversight from when I extracted everything to a hook. Good catch.